### PR TITLE
chore: update mergify

### PR DESCRIPTION
**Pull request by mpoke**
- remove v7 & v10 backports
- add author as backport assignee

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -45,12 +42,3 @@ pull_request_rules:
       backport:
         branches:
           - release/v9.0.x
-
-  - name: Backport patches to the release/v10.0.x branch
```
Comment (glnro) [inline .mergify.yml]: Why remove 10 if we'll need to add it back soon?

Comment (mpoke) [reply .mergify.yml]: Good point. I added it back.

**Review by sainoe — APPROVED** [review]

File: .mergify.yml [hunk 2]
```diff
@@ -18,15 +24,6 @@ pull_request_rules:
...
-  - name: Backport patches to the release/v7.0.x branch
-    conditions:
-      - base=main
-      - label=A:backport/v7.0.x
-    actions:
-      backport:
-        branches:
-          - release/v7.0.x
-
   - name: Backport patches to the release/v8.0.x branch
```
Comment (mmulji-ic) [inline .mergify.yml:27]: How about the following condition for all backports?
```
    conditions:
      - "#approved-reviews-by>=1"
```

Comment (mpoke) [reply .mergify.yml:27]: It's already part of the branch protection rule. That will overwrite whatever you add here.

**Review by mmulji-ic — APPROVED** [review]

**Pull request merged**